steps/prepare_release.sh

```diff
@@ -69,6 +72,9 @@ if [[ ${#ARR[@]} -gt 3 ]]; then
     # refer: https://www.python.org/dev/peps/pep-0440/#pre-releases
     PY_VERSION="${ARR[0]}.${ARR[1]}.${ARR[2]}${ARR[3]}"
 fi
+if [[ ${VERSION} =~ "SNAPSHOT" ]]; then
+    PY_VERSION="${ARR[0]}.${ARR[1]}.${ARR[2]}a0"
```

Comment:
conflict with line 73. I intend to run as :
```sh
./prepare_release.sh 1.2.3.a0
```

then the `PY_VERSION` is `1.2.3.a0`. For cpp, `.a0` is not token; for java, `.a0` always replaced with `-SNAPSHOT`.  See the comment at top of the file. 

See if you require more semantic, or we can revert the change.